### ISSUE TITLE
Ensemble selection is Gumbel-logit: name the temperature and make it calibration-adaptive

### ISSUE THREAD

**Issue by microprediction** [microprediction/skaters#213]
## The observation

The ensemble weighting already is Gumbel-logit selection, we just don't name it or use the freedom it gives.

In `bayesian.py` and `terminal.py` the candidate weight is a softmax over accumulated, learning-rate-scaled log-evidence:

```
log_w[i] += learning_rate * lp - complexity_penalty * depths[i]      # per candidate
w[i]      = exp(log_w[i] - max_lw)                                    # softmax
```

Under random-utility maximization, `softmax(beta * score)` is exactly the choice probability when the candidate scores carry i.i.d. Gumbel noise of scale `1/beta`. So `learning_rate` is playing the role of an inverse temperature `beta`, and `forget` (in `terminal.py`) bounds how much evidence accumulates before the softmax. This is the tractable case McFadden built discrete choice on: Gumbel is the one noise family for which the argmax-selection integral collapses to a closed form. The Gaussian is the analogous tractable case for the additive side of the tree (leaves, transforms, `Dist.combine`), and the GPD is the tractable case for the tail splice. The selection operation has its own conjugate-like family and it is already the one we use.

Naming it is not the point. The point is that `beta` is currently fixed and implicit, and there are two things the framing hands us that the code does not exploit.

## 1. An explicit, calibration-adaptive temperature

`learning_rate` fixes the peakedness of the pool for all time. But the right temperature is regime-dependent:

- when one candidate genuinely dominates and is calibrated, we want to run cold (commit to it),
- under a regime shift or global misspecification, every candidate is wrong and we want to run hot (hedge across the pool) until the evidence re-sorts.

A fixed `beta` cannot do both. The signal that should drive `beta` is already computed elsewhere in the stack: realized one-step calibration. When the mixture's PIT drifts off uniform or the running log-loss degrades, heat up; when PIT is uniform and log-loss is low, cool down. Concretely, adapt `beta` online toward a target predictive dispersion, the same lever `parade`/`residual_transform` already use for scale, but applied to the selection temperature rather than the leaf variance.

Proposal: expose `temperature` (or `beta`) as a named parameter on `bayesian_ensemble` and `terminal_leaf_ensemble`, defaulting to the current behavior (`beta = learning_rate`, constant), and add an opt-in adaptive mode driven by a PIT/log-loss statistic.

## 2. A sampling path for coherent trajectories (only if the spatial/multivariate direction goes anywhere)

Averaging candidate Dists is correct for a point or marginal predictive. But if we ever want a *diverse ensemble of full trajectories* (the spatiotemporal-mesh direction, where a blurred per-pixel mean is exactly the wrong output), the Gumbel-max trick lets us sample a candidate in one pass with no normalization: add i.i.d. Gumbel(0, 1/beta) to each `log_w[i]` and take the argmax. Repeated draws give coherent samples from the selection distribution instead of a variance-inflating average. This is speculative and gated on whether the multivariate work happens; noting it so the connection is on record.

## Cheap experiment for (1)

Against the existing `laplace` pool on the standard benchmark set:

1. sweep a fixed `beta` and confirm the current default is not already optimal per series (it very likely is not, since the pool's dispersion should differ across regimes),
2. add the adaptive rule and measure mean log-score and PIT L1 against the best fixed `beta`,
3. the win condition is the usual one: log-score up **and** calibration held.

This is a small, self-contained study, no new distributions, no new machinery, just making an implicit constant into an adaptive scalar with a principled name. The recalibration angle is the same one surfacing in the spatiotemporal-mesh probe, so a positive result here also de-risks that direction.

## Prior art

McFadden 1974 (conditional logit) and the 2000 Nobel; the Gumbel/logit/Luce-axiom equivalence in Yellott 1977, with Luce and Marley earlier. None of this is new theory; the value is connecting it to the ensemble knob we already ship.
